src/System.IO.Pipelines/System/IO/Pipelines/Pipe.cs

```diff
@@ -14,6 +14,8 @@ namespace System.IO.Pipelines
     /// </summary>
     public class Pipe : IPipe, IPipeReader, IPipeWriter, IReadableBufferAwaiter, IWritableBufferAwaiter
     {
+        private const int SegmentPoolSize = 16;
```

Comment:
How did you come up with this?

For Kestrel, the 16 pooled segments could reference upwards of 64KB. This is just enough for the response buffer, but it's possible the request buffer could need unpooled segments if the app falls too far behind reading it.

Reply:
Selected randomly not to be too large. I think optimizing this allocation matters more in high performance scenarios where reader would either would be fast enough to keep up or request/response sizes would not be too large.